api/src/convert/block.rs

```diff
@@ -31,6 +38,12 @@ impl TryFrom<&blockchain::Block> for Block {
         let root_element = TxOutMembershipElement::try_from(value.get_root_element())?;
         let contents_hash = BlockContentsHash::try_from(value.get_contents_hash())?;
 
+        let timestamp = if value.get_version().timestamps_are_supported() {
+            value.get_timestamp()
+        } else {
+            0
+        };
```

Comment:
Should we error out if we got a timestamp in a version less than 3, or is zeroing an ok thing to do here?

Reply:
I actually am not sure we should have the check here at all, since its not like we validate the block_id is correct either, but if we were to keep the check then I think erroring is the right move. I don't think we should have logic in conversion methods that modify the data.

Reply:
Are you thinking remove these and place a panic where we create the block?

Then the test utilities would have the logic to ensure tests populate timestamps appropriatly?

Reply:
> Are you thinking remove these and place a panic where we create the block?
> 
> Then the test utilities would have the logic to ensure tests populate timestamps appropriatly?

Yes thats what I was thinking